### PR TITLE
 fix: fix circular sort in react-flicking

### PR DESCRIPTION
**Pull request by daybrush**
## Issue

#172

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -206,4 +206,5 @@ report/
 temp/
 demo/release
 demo/_data/version.yml
-dist/
+packages/*/dist/
+dist/
```
Comment (WoodNeck) [inline .gitignore]: Does `editorconfig` is working correctly?

File: packages/react-flicking/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@egjs/react-flicking",
-  "version": "0.0.4",
+  "version": "0.0.6",
```
Comment (WoodNeck) [inline packages/react-flicking/package.json]: Why it isn't `0.0.5`?

Comment (daybrush) [reply packages/react-flicking/package.json]: @WoodNeck 
 It seems that version 0.0.5 was missing when push was previously pressed.

This version has already been distributed.

File: packages/react-flicking/src/react-flicking/index.tsx [hunk 3]
```diff
@@ -77,13 +77,15 @@ export default class Flicking extends React.Component<IFlickingProps> {
...
     panel.count = panel.$list.length;
-    panel.origCount = panel.count;
+    panel.origCount = [].concat(this.props.children).length;
 
     if (this.props.circular) {
       flicking._arrangePanels();
-      flicking._movePanelPosition(panel.no - panel.index, true);
+      const deleteCount = panel.no - panel.index;
+      const isAppend = deleteCount >= 0;
+      flicking._movePanelPosition(isAppend ? deleteCount : -deleteCount, isAppend);
```
Comment (WoodNeck) [inline packages/react-flicking/src/react-flicking/index.tsx]: What does above code means?

Comment (daybrush) [reply packages/react-flicking/src/react-flicking/index.tsx]: @WoodNeck 
panel.no is the index of the panel in the position indicated by Camera.

panel.index is the current panel.

The positions are set in the order of the list, but the alignment is initialized each time you renew.

(panel.no - panel to align.send back or forward as much as index.

**Comment by jongmoon** [comment]
PR refer to `react-infinitegrid` not `react-flicking`.

File: packages/react-flicking/src/react-flicking/index.tsx (hunk 3 above)
Comment (jongmoon) [inline packages/react-flicking/src/react-flicking/index.tsx]: @daybrush Is it needed to call `_movePanelPosition` when deleteCount is `0`?

File: packages/react-flicking/src/react-flicking/index.tsx [hunk 4]
```diff
@@ -77,13 +77,15 @@ export default class Flicking extends React.Component<IFlickingProps> {
     const panel = flicking._conf.panel;
     const horizontal = this.props.horizontal;
     const className = `${this.props.prefix}-panel`;
-    panel.$list = Array.prototype.slice.call(flicking.$container.children);
+    panel.$list = [].slice.call(flicking.$container.children);
```
Comment (jongmoon) [inline packages/react-flicking/src/react-flicking/index.tsx]: @daybrush Why changed?

File: packages/react-flicking/src/react-flicking/index.tsx [hunk 5]
```diff
@@ -77,13 +77,15 @@ export default class Flicking extends React.Component<IFlickingProps> {
...
     panel.count = panel.$list.length;
-    panel.origCount = panel.count;
+    panel.origCount = [].concat(this.props.children).length;
```
Comment (jongmoon) [inline packages/react-flicking/src/react-flicking/index.tsx]: @daybrush Could it be panel.$list different from `this.props.children`?

Comment (daybrush) [reply packages/react-flicking/src/react-flicking/index.tsx]: @jongmoon 
$list is a panel list with clones.
origCount is the number of original panels except for clones.

Comment (jongmoon) [reply packages/react-flicking/src/react-flicking/index.tsx]: @daybrush I got it!

File: packages/react-flicking/src/react-flicking/index.tsx [hunk 6]
```diff
@@ -77,13 +77,15 @@ export default class Flicking extends React.Component<IFlickingProps> {
...
 
     if (this.props.circular) {
       flicking._arrangePanels();
-      flicking._movePanelPosition(panel.no - panel.index, true);
+      const deleteCount = panel.no - panel.index;
```
Comment (jongmoon) [inline packages/react-flicking/src/react-flicking/index.tsx]: @daybrush moveCount

**Pull request merged**